### PR TITLE
statefulset: add Delete() func

### PR DESCRIPTION
**Pull request by sebrandon1**
Adds the functionality and the unit test for a Delete() func in the statefulset package.

### CODE REVIEW

File: pkg/statefulset/statefulset.go [hunk 1]
```diff
@@ -205,6 +205,30 @@ func (builder *Builder) Exists() bool {
...
+// Delete a statefulset from the cluster.
+func (builder *Builder) Delete() error {
+	if valid, err := builder.validate(); !valid {
+		return err
+	}
+
+	glog.V(100).Infof("Deleting statefulset %s in namespace %s", builder.Definition.Name, builder.Definition.Namespace)
+
+	if !builder.Exists() {
+		return nil
```
Comment (kononovn) [inline pkg/statefulset/statefulset.go:217]: Add glog message

Comment (sebrandon1) [reply pkg/statefulset/statefulset.go:217]: Other Delete() methods don't use a glog for a resource not existing:
- [deployment](https://github.com/openshift-kni/eco-goinfra/blob/main/pkg/deployment/deployment.go#L430-L434)
- [daemonset](https://github.com/openshift-kni/eco-goinfra/blob/main/pkg/daemonset/daemonset.go#L293-L297)
- [configmap](https://github.com/openshift-kni/eco-goinfra/blob/main/pkg/configmap/configmap.go#L127-L129)

**Review by klaskosk — APPROVED** [review]
/lgtm

File: pkg/statefulset/statefulset.go (hunk 1 above)
Comment (kononovn) [reply pkg/statefulset/statefulset.go:219]: That has to be improved. Please refer to: https://github.com/openshift-kni/eco-goinfra/blob/411277cec65bd7a730c2bdf9938ad61557eca100/pkg/nmstate/nmstate.go#L131

**Review by kononovn — CHANGES_REQUESTED** [review]
Please add glog message as in example above.

File: pkg/statefulset/statefulset.go (hunk 1 above)
Comment (sebrandon1) [reply pkg/statefulset/statefulset.go:221]: Done!

**Review by klaskosk — APPROVED** [review]

**Review by kononovn — APPROVED** [review]
/lgtm

**Pull request merged**